### PR TITLE
Assistant

### PR DESCRIPTION
**Pull request by ar7nic**

### CODE REVIEW

File: assistants/assistants.js [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
-   static cartAssistant = new CartAssistant();
-   static orderAssistant = new OrderAssistant();
-}
+// class Assistants {
+//     loginAssistant = new LoginAssistant();
+//     cartAssistant = new CartAssistant();
+//     orderAssistant = new OrderAssistant();
+// }
 
+export const ASSISTANT = {
```
Comment (serhii-lugo) [inline assistants/assistants.js]:
```suggestion
export const ASSISTANT = {
```
ASSISTANT in plural

File: assistants/cartAssistant.js [hunk 2]
```diff
@@ -1,22 +1,58 @@
...
+import { expect } from "@playwright/test";
+import { PAGES } from "../pages/pages";
+import { ASSISTANT, Assistants } from "./assistants";
 
 export class CartAssistant {
+  async popUpAccept(page) {
+    page.on("dialog", async (dialog) => {
+      await dialog.accept();
+    });
+  }
```
Comment (serhii-lugo) [inline assistants/cartAssistant.js:9]:
```suggestion
  async popUpAccept(page) {
    page.on("dialog", async (dialog) => {
      await dialog.accept();
    });
  }
```

can be moved to some general type of assistant like UserAssistant.

File: assistants/cartAssistant.js [hunk 3]
```diff
@@ -1,22 +1,58 @@
...
-    async addToCartFirstItem(page){
-        await page.waitForLoadState('networkidle');
-        await page.locator(PAGES.categoriesPage.itemsTitles).first().click();
-        await page.waitForLoadState('networkidle');
-        await page.locator(PAGES.productPage.addToCartBtn).click();
-        await Assistants.cartAssistant.popUpAccept(page,'Product added');
+    await page.locator(PAGES.productPage.addToCartBtn).click();
+    await ASSISTANT.cartAssistant.popUpAccept(page);
+    await page.waitForTimeout(1000);
+    return { prodName, prodPrice };
```
Comment (serhii-lugo) [inline assistants/cartAssistant.js:24]:
```suggestion
    return { prodName, prodPrice };
```
good solution!

File: assistants/cartAssistant.js [hunk 4]
```diff
@@ -1,22 +1,58 @@
...
-        })
-    }
-    async addToCartFirstItem(page){
-        await page.waitForLoadState('networkidle');
-        await page.locator(PAGES.categoriesPage.itemsTitles).first().click();
-        await page.waitForLoadState('networkidle');
-        await page.locator(PAGES.productPage.addToCartBtn).click();
-        await Assistants.cartAssistant.popUpAccept(page,'Product added');
+    await page.locator(PAGES.productPage.addToCartBtn).click();
+    await ASSISTANT.cartAssistant.popUpAccept(page);
```
Comment (serhii-lugo) [inline assistants/cartAssistant.js]:
```suggestion
    await ASSISTANT.cartAssistant.popUpAccept(page);
```

you can use -> await thist.popUpAccept(page);

File: assistants/orderAssistant.js [hunk 5]
```diff
@@ -1,22 +1,32 @@
-import {PAGES} from "../pages/pages";
-import {Assistants} from "./assistants";
-import {expect} from "@playwright/test";
-
+import { PAGES } from "../pages/pages";
+import { USERS } from "../models/users";
 
 export class OrderAssistant {
+  async placeTheOrder(page) {
```
Comment (serhii-lugo) [inline assistants/orderAssistant.js:5]:
```suggestion
  async placeTheOrder(page) {
```

you should put user object as parameter to this method. And use user object inside method.
      -> 
 async placeTheOrder(page, userObj) {
                    await page.locator(PAGES.orderPopup.city).fill(userObj.userCity);
  }

File: assistants/signInAssistant.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import { PAGES } from "../pages/pages";
+
+export class SignInAssistant {
+  async signInToSite(page, name, password) {
+    await page.locator(PAGES.mainMenu.signInMenu).click();
+    await page.locator(PAGES.signInPopUp.signInUserNameInput).fill(name);
+    await page.locator(PAGES.signInPopUp.signInPasswordInput).fill(password);
+    await page.locator(PAGES.signInPopUp.signUpButton).click();
+  }
```
Comment (serhii-lugo) [inline assistants/signInAssistant.js:9]:
```suggestion
export class SignInAssistant {
  async signInToSite(page, name, password) {
    await page.locator(PAGES.mainMenu.signInMenu).click();
    await page.locator(PAGES.signInPopUp.signInUserNameInput).fill(name);
    await page.locator(PAGES.signInPopUp.signInPasswordInput).fill(password);
    await page.locator(PAGES.signInPopUp.signUpButton).click();
  }
```

you have loginAssistant for this case..

File: package.json [hunk 7]
```diff
@@ -9,5 +9,8 @@
   "devDependencies": {
     "@playwright/test": "^1.29.1"
   },
-  "scripts": {}
+  "scripts": {},
+  "dependencies": {
+    "prettier": "^2.8.3"
```
Comment (serhii-lugo) [inline package.json:14]:
```suggestion
    "prettier": "^2.8.3"
```

good choice !

**Review by serhii-lugo — CHANGES_REQUESTED** [review]
need to do small update

File: assistants/signInAssistant.js (hunk 6 above)
Comment (ar7nic) [reply assistants/signInAssistant.js:9]: but they are different popups with their own locators.  how can I use it?

Comment (serhii-lugo) [reply assistants/signInAssistant.js:9]: yep. you are right. My bad!

**Pull request merged**